### PR TITLE
Add MongoDB connector

### LINKED ISSUE #14: Implement a MongoDB connector

### PR DESCRIPTION
**Pull request by LotuxPunk**
***Currently WIP*** ⚠️

### CODE REVIEW

File: packages/ldes-mongodb-connector/lib/MongoDBConnector.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    } catch {
+      console.log(member);
+    }
+
+    const collection = this.db.collection('ldes');
+
+    let id = JSONmember['@id'];
+    let type = JSONmember['@type'];
+    let time = JSONmember['http://www.w3.org/ns/prov#generatedAtTime'];
+    let isVersionOf = JSONmember['http://purl.org/dc/terms/isVersionOf']['@id'];
```
Comment (pietercolpaert) [inline packages/ldes-mongodb-connector/lib/MongoDBConnector.ts]: This should be part of a generic solution to create a version materialization of an LDES. dcterms:isVersionOf is just one way of having a link to a higher level version thing. We could for example also think about `all the latest air quality observations sosa:madeBy a sensor`.

 I think the first focus should be to store _all_ the version objects as they are, and not yet do the version materialization, which probably should be a feature of the LDES client itself.

Related issue: https://github.com/TREEcg/event-stream-client/issues/13

Comment (LotuxPunk) [reply packages/ldes-mongodb-connector/lib/MongoDBConnector.ts]: Absolutely, we have to look at this!

I think that this storage takes part in the discovery of what is possible.
Storing isVersionOf allows us to have a granularity in what is kept or not (number of version to keep, etc.)

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,6 @@
+root = true
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+charset = utf-8
```
Comment (DemianD) [inline .editorconfig]: Add a newline

File: packages/ldes-mongodb-connector/__tests__/ldes-mongodb-connector.test.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+'use strict';
```
Comment (DemianD) [inline packages/ldes-mongodb-connector/__tests__/ldes-mongodb-connector.test.js:1]: I'd remove this file for now.

Comment (DemianD) [reply packages/ldes-mongodb-connector/__tests__/ldes-mongodb-connector.test.js:1]: Maybe create an issue to create some integration tests

File: packages/ldes-mongodb-connector/lib/MongoDBConnector.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import type { IWritableConnector } from '@ldes/types';
+// import mongoose, { Schema } from 'mongoose';
```
Comment (DemianD) [inline packages/ldes-mongodb-connector/lib/MongoDBConnector.ts]: Can be removed

File: packages/ldes-mongodb-connector/lib/MongoDBConnector.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+import { MongoClient } from 'mongodb';
+
+export class MongoDBConnector implements IWritableConnector {
+  private readonly config: any;
+  private db: any;
```
Comment (DemianD) [inline packages/ldes-mongodb-connector/lib/MongoDBConnector.ts]: A lot of any. I'll have a look at it

File: packages/ldes-mongodb-connector/lib/MongoDBConnector.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  public constructor(config: any) {
+    this.config = config;
+  }
+
+  /**
+   * Writes a version to the corresponding backend system
+   * @param member
+   */
+  public async writeVersion(member: any): Promise<void> {
+    var JSONmember;
```
Comment (DemianD) [inline packages/ldes-mongodb-connector/lib/MongoDBConnector.ts]: const/let

File: packages/ldes-mongodb-connector/lib/MongoDBConnector.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    try {
+      JSONmember = JSON.parse(member);
+    } catch {
+      console.log(member);
```
Comment (DemianD) [inline packages/ldes-mongodb-connector/lib/MongoDBConnector.ts:26]: If there is an error, it'll still continue with the code below

File: packages/ldes-mongodb-connector/lib/MongoDBConnector.ts [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+      await client.connect();
+      // Establish and verify connection
+      this.db = await client.db('ldes');
+      console.log('Connected successfully to server');
+    } catch (err: any) {
+      console.log(err);
+    } finally {
+      // Ensures that the client will close when you finish/error
+      //await client.close();
+    }
```
Comment (DemianD) [inline packages/ldes-mongodb-connector/lib/MongoDBConnector.ts]: I'd remove this section.

**Review by DemianD — COMMENTED** [review]
Really nice done already! Some comments/nitpicks on the code. I haven't tested it out functionally.

File: packages/ldes-mongodb-connector/lib/MongoDBConnector.ts (hunk 1 above)
Comment (DemianD) [reply packages/ldes-mongodb-connector/lib/MongoDBConnector.ts]: I created an issue to keep track of this: https://github.com/osoc21/ldes2service/issues/20

**Review by DemianD — APPROVED** [review]

**Pull request merged**